students/rachel_s/session04/kata14.py

```diff
@@ -24,11 +37,51 @@ def build_trigrams(words):
         if not pair in trigrams:
             trigrams[pair] = [follower]
         else:
-            trigrams[pair] += [follower]
-
+            trigrams[pair].append(follower)
     return trigrams
 
 
+def build_text(word_pairs):
+    output = []
+    for i in range(10):
+        starter = start_sentence()
+        make_sentence(output, starter)
+    
+    return output
+
+def start_sentence():
+    starters = list(filter(lambda tup: tup[0] == ".",  word_pairs.keys()))
```

Comment:
Great to see functional `filter(...)` and `lambda`!

Now you learned list comprehension, you may use it for this kind of occasion in the future... :)

I'm a little unsure, though, about the filtering condition (`lambda tup: tup[0] == "."`). Does this mean that some of your pairs might have `"."` as its first part? I didn't check your regex carefully, so I suppose it might be possible, but then I suppose the desired intention was to get rid of all the punctuation, not retain them. Well, there's no hard requirement for that, so I suppose this is fine.